### PR TITLE
Add support for routine control types.

### PR DESCRIPTION
**Pull request by p-dial8891**
Purpose: 
Could not find any code for 0x31 routine services for UDS. This pull request could be a start to get more UDS api.

Tests: 
Performed a build locally to see if it compiled - OK. 
No tests done with hardware or even virtual CAN sockets.

### CODE REVIEW

**Comment by nyurik** [comment]
@p-dial8891 thanks!  Could you add a doc comment to the `RoutineControlType` explaining which spec it follows? E.g. a page number or a chapter name that describes this feature in the ISO docs? Or anything else similar?  Thx!

**Comment by nyurik** [comment]
P.S. Make sure to run `cargo fmt --all`

**Comment by nyurik** [comment]
P.P.S. please also add a comment to the https://github.com/nyurik/automotive_diag/blob/7678163a06132e9660b704c001258ff529f5e5d0/src/uds/commands.rs#L52 -- with a link to the new enum

File: src/uds/routine.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+use crate::enum_wrapper;
+use enum2repr::EnumRepr;
+
+enum_wrapper!(uds, RoutineControlType, RoutineControlTypeByte);
+
+/// UDS Routine (0x31) service control types.
```
Comment (nyurik) [inline src/uds/routine.rs:6]:
```suggestion
/// UDS Routine (0x31) service control types.
/// See chapter `14.2 RoutineControl service` in the ISO 14229 spec.
```

**Comment by p-dial8891** [comment]
Hi @nyurik,
Well regarding the reference, I do not have access to the UDS spec at the moment and I wont have it in the future. What I did was I referred, Services — udsoncan 1.17.1 documentation<https://udsoncan.readthedocs.io/en/latest/udsoncan/services.html#routinecontrol-0x31>, which is another project, just to confirm what I knew from previous UDS implementations I have worked on in the past. Are you ok referencing another similar open source project ? If there is a copy of the ISO spec that I could access, I could go through it as I have used it in the past.

Thank you for your quick replies.

________________________________

**Comment by p-dial8891** [comment]
I have to admit I am not sure how to add a link to an enum in the comment. I assume you are saying to add github links in the comments, but somehow I think that is not what you meant.

Could you give a short example?

-proteus
________________________________

**Comment by nyurik** [comment]
I'll fix the link, don't worry. Please do the cargo fmt

**Comment by nyurik** [comment]
thanks again, merged, will address the docs separately (there are some issues with it)

**Pull request merged**